gcloud/datastore/key.py

```diff
@@ -431,25 +432,25 @@ def _dataset_ids_equal(dataset_id1, dataset_id2):
     (Ties are resolved since 'foo' can only be an alias for one of
     s~foo or e~foo in the backend.)
 
-    :type dataset_id1: string
-    :param dataset_id1: A dataset ID.
+    :type project1: string
+    :param project1: A project.
 
-    :type dataset_id2: string
-    :param dataset_id2: A dataset ID.
+    :type project2: string
+    :param project2: A project.
 
     :rtype: boolean
-    :returns: Boolean indicating if the IDs are the same.
+    :returns: Boolean indicating if the projects are the same.
     """
-    if dataset_id1 == dataset_id2:
+    if project1 == project2:
         return True
 
-    if dataset_id1.startswith('s~') or dataset_id1.startswith('e~'):
-        # If `dataset_id1` is prefixed and not matching, then the only way
-        # they can match is if `dataset_id2` is unprefixed.
-        return dataset_id1[2:] == dataset_id2
-    elif dataset_id2.startswith('s~') or dataset_id2.startswith('e~'):
-        # Here we know `dataset_id1` is unprefixed and `dataset_id2`
+    if project1.startswith('s~') or project1.startswith('e~'):
```

Comment:
Yeah one of the greatest lines in https://cloud.google.com/datastore/v1beta3/release-notes is

> Field `dataset_id` is now `project_id`. `project_id` never includes the app/project ID "prefix" (e.g. `s~`). This fixes https://github.com/GoogleCloudPlatform/google-cloud-datastore/issues/59.